### DISCUSSION TITLE
utztime - Time and TimeZone Library

### DISCUSSION THREAD

**Discussion by shaneapowell** [micropython/micropython#14321] [Show and tell]
Hi Guys.
I do a good amount of my projects on ESP32 devices, with NTP enabled.
I always need a good way to handle timezones and safe time math.
In regular Python, I tend to prefer Arrow.   I couldn't find anything that quite met my needs in the micropython world.
So.. I created my own!   I'v e been using this with success so far on one of my personal side-projects (still in early development).
But, I thought it was long over due for me to give back.

https://github.com/shaneapowell/utztime/

Cheers
Shane.

**Comment by Josverl** [comment]
It would be more useful to find a way to cover more than a single, although large, country.

Could you refactor to allow more 'chunks' of the world to be used separately?

**Comment by shaneapowell** [comment]
I totaly agree.  I'll come up with a good memory-efficent way of easily adding more timezones.  As well as a clean unit-test mechanism for them too.  Appreciate the feedback.

**Comment by shaneapowell** [comment]
Well.. that was a fun adventure today.
Made a good number of changes to help in support adding timezones.
Unfortunatly, I didn't actually add any new zones  yet though.  
I did however, better organize the existing timezones, and made it easier for additional timezones, and regions to be added.
People using this library can create their own timezones, and register them into the tzlist module.
And.. anybody wanting to have additional timezones added to the library, should be able to easily(ish) create the definitions and create a PullRequest to have them added to the main code.
